root/usr/lib64/ruby/vendor_ruby/discovery/menu.rb

```diff
@@ -84,6 +84,7 @@ def configure_network static, mac, ip=nil, gw=nil, dns=nil
   command("nmcli connection reload")
   command("nmcli connection down primary")
   result = command("nmcli connection up primary", false)
+  command("nm-online -s -q --timeout=45") unless static
```

Comment:
This method is called from TUI on all network configuration actions, this actually makes sure its online when using DHCP, so proxy start on the very next line won't fail.